src/main/java/org/openrewrite/java/migrate/javax/AddDefaultConstructorToEntityClass.java

```diff
@@ -0,0 +1,83 @@
+    @Override
+    public String getDisplayName() {
+        return "`@Entity` objects with constructors must also have a default constructor";
+    }
+
+    @Override
+    public String getDescription() {
+        return "When a Java Persistence API (JPA) entity class has a constructor with arguments, the class must also " +
+               "have a default, no-argument constructor. The OpenJPA implementation automatically generates the " +
+               "no-argument constructor, but the EclipseLink implementation does not.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                Preconditions.or(
+                        new UsesType<>("javax.persistence.Entity", true),
+                        new UsesType<>("javax.persistence.MappedSuperclass", true)),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                        // Exit if class not annotated with either @Entity or @MappedSuperclass
+//                        if (FindAnnotations.find(classDecl, "javax.persistence.Entity").isEmpty()
+//                            && FindAnnotations.find(classDecl, "javax.persistence.MappedSuperclass").isEmpty()) {
+                        if (classDecl.getLeadingAnnotations().stream()
+                                .noneMatch(anno ->
+                                    anno.getType().toString().equals("javax.persistence.Entity")
+                                    || anno.getType().toString().equals("javax.persistence.MappedSuperclass"))) {
+                            return super.visitClassDeclaration(classDecl, ctx);
+                        }
+
+                        // Exit if class already has default no-arg constructor
+                        if (classDecl.getBody().getStatements().stream()
+                                .filter(statement -> statement instanceof J.MethodDeclaration)
+                                .map(J.MethodDeclaration.class::cast)
+                                .filter(J.MethodDeclaration::isConstructor)
+                                .anyMatch(constructor -> constructor.getParameters().get(0) instanceof J.Empty)) {
+                            return super.visitClassDeclaration(classDecl, ctx);
+                        }
+
+                        // Add default constructor with empty body
+                        return classDecl.withBody(JavaTemplate.builder("public #{}(){}")
+                                .contextSensitive()
+                                .build()
+                                .apply(new Cursor(getCursor(), classDecl.getBody()),
+                                        classDecl.getBody().getCoordinates().addMethodDeclaration(Comparator.comparing(J.MethodDeclaration::getSimpleName)),
```

Comment:
Noticing intermittent discrepencies on the insertion order for `addMethodDeclaration` using this comparator, which causes intermittent unit test failures. 

Maybe should just pick a consistent insertion point for tests? Even if the resulting code might not look as organized...

Reply:
Switched to using `firstStatement` to guarantee order for tests.